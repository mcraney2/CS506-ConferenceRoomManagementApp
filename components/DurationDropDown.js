//import DropDownPicker from 'react-native-dropdown-picker';
import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import {Picker} from '@react-native-picker/picker';

class DurationDropDown extends Component {
    constructor(props) {
        super(props);
        this.state = {  
        
        }
    }
    render() { 
        return (  
            <>
                <View style = {styles.container}>
                
                    <Picker
                        selectedValue = {this.props.hours}
                        onValueChange = {this.props.setHours}
                        testID = 'Hours'
                        style={{ height: 25, width: 150 }}
                        >
                            <Picker.Item label = '0 hours' value = '0' />
                            <Picker.Item label = '1 hour' value = '1' />
                            <Picker.Item label = '2 hours' value = '2' />
                            <Picker.Item label = '3 hours' value = '3' />
                        </Picker>
                    <Picker
                        selectedValue = {this.props.minutes}
                        onValueChange = {this.props.setMinutes}
                        style={{ height: 25, width: 150 }}
                        testID = 'Minutes'
                        >
                            <Picker.Item label = '0 minutes' value = '0' />
                            <Picker.Item label = '15 minutes' value = '15' />
                            <Picker.Item label = '30 minutes' value = '30' />
                            <Picker.Item label = '45 minutes' value = '45' />
                            </Picker>
                    </View>

</>
        );
    }
}
const styles = StyleSheet.create({
    container: {
      flex: 1,
      flexDirection: 'row',
      alignItems: 'center',
      //justifyContent: 'center',
    },
    headerText: {
        fontSize: 22,
        fontWeight: 'bold'
    }
})
export default DurationDropDown;

