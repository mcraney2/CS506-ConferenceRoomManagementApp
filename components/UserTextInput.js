import React, { Component } from 'react';
import { TextInput } from 'react-native-gesture-handler';

class UserTextInput extends Component {
    constructor(props) {
        super(props);
        this.state = {  
            height: 30,
            
            width: 250,
            borderColor: "grey",
            borderWidth: 1,
            margin: 10,
        }
    }
    updateSize = (height) => {
        this.setState({
          height
        });
      }
    render() { 
        const newValue = this.props.value;
        const height = this.state;
        let newStyle = height;
        return (  
        <TextInput
            placeholder= {this.props.placeHolder}
            onChangeText={(value) => this.props.setValue(value)}
            style={newStyle}
            editable={true}
            multiline={true}
            value={newValue}
            testID = 'text'
            onContentSizeChange={(e) => this.updateSize(e.nativeEvent.contentSize.height)}
          />);
    }
}
 
export default UserTextInput;