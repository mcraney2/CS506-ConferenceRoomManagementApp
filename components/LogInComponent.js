import React, { Component } from 'react';
import { Text, TouchableOpacityBase, View,StyleSheet} from 'react-native';

import RoomSelectDropdown from '../components/RoomSelectDropdown'
import UserTextInput from '../components/UserTextInput'
import Button from '../components/Button'
import axios from 'axios'
import { TextInput } from 'react-native-gesture-handler';
import RepeatSelectDropdown from './RepeatSelectDropdown';
import { Calendar } from 'react-native-big-calendar';

class LogInComponent extends Component {
    
    constructor(props) {
        super(props);
        this.state = { 
            username: '',
            password: '',
            response: ''
        }
    }
    static navigationOptions = {
        title : 'Home',
    };

    setUsername (newUsername) {
        this.setState({username : newUsername});
    }

    setPassword (newPassword) {
        this.setState({password : newPassword});
    }

    sendRequest(username, password) {
        console.log(username, password);
        const request = JSON.stringify(
          { 
            username: username.value,
            password: password.value

        });
        axios.get('http://10.0.2.2:8000/room_mgmt/login', request) 
        .then(function (response) {
          this.state.response = response;
          // Now based on authentication/accoutn status decide where to navigate
          // Authentication is false, deny log-in so do nothing ?
          if (this.state.response['authenticated'] == false) {

          }
          // Authentication is true
          else {
              // Redirect to managment console
              if(this.state.response['accountType'] == 'admin') {
                    this.props.navigation.navigate('ManagementConsole')
              }
              // Redirect to user console
              else {
                    this.props.navigation.navigate('UserConsole');
              }
          }
        })
        .catch(function (error) {
          console.log(error);
        });
    }

    render() { 
        return ( 
            <>
            <View>
                <Text styles={styles.textSty3}>Log-In Information</Text>
                <UserTextInput placeHolder = 'Username' value = {this.state.username} setValue = {this.setUsername.bind(this)}/>
                <UserTextInput placeHolder = 'Password' value = {this.state.password} setValue = {this.setPassword.bind(this)}/>
                <Button 
                        // Make this the line the replacement when everything has been handled
                        // handleClick = {() => this.sendRequest(this.state.username, this.state.password)}
                        handleClick = {() => this.props.navigation.navigate('Management Console')}
                        label = "Log-In"
                    />
            </View>
            </>
        );
    }
}
const styles = StyleSheet.create({
    input: {
        paddingRight: 10,
        lineHeight: 23,
        flex: 1,
        textAlignVertical: 'top'
    },
    container: {
        flex: 0.8,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
      },
    container2: {
        margin:10,
        flex:1,
        width:350,
    },
    sideBySide: {
        flex:0.3,
        margin:10,
        flexDirection:'row',
    },
    button: {
        margin: 10,
    },
    textSty3: {
        fontSize:20,
        fontWeight:'bold',
        marginTop: 10,
        marginRight: 10,
        marginBottom: 10,
        marginLeft: 10,
        color:'#474747',
    },
})

export default LogInComponent;