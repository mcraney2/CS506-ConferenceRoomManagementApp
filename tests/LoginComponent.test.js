import React from 'react';
import renderer from 'react-test-renderer';
import  DurationDropDown  from '../components/DurationDropDown';
import { fireEvent, render } from '@testing-library/react-native';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import LogInComponent from '../components/LogInComponent'
import { TestScheduler } from '@jest/core';
import UserTextInput from '../components/UserTextInput';


test('Renders LogInComponent snapshot as expected', () => {
    const tree = renderer.create(<LogInComponent/>).toJSON();
    expect(tree).toMatchSnapshot();
});



test('Username text input sends username text to event handler', () => {
    const onEventMock = jest.fn();
    const {getByTestId} = render(
        <UserTextInput placeHolder = '' setValue = {onEventMock}/>
    );
    const answer1 = "user1"
    fireEvent(getByTestId('text'), 'onChangeText', 'user1');
    expect(onEventMock).toHaveBeenCalledWith(answer1);
});

test('Password text input sends password text to event handler', () => {
    const onEventMock = jest.fn();
    const {getByTestId} = render(
        <UserTextInput placeHolder = '' setValue = {onEventMock}/>
    );
    const answer1 = "password123"
    fireEvent(getByTestId('text'), 'onChangeText', 'password123');
    expect(onEventMock).toHaveBeenCalledWith(answer1);
});

test('Username handler sets the state of username', () => {
    Enzyme.configure({adapter: new Adapter()})
    const test = shallow(<LogInComponent/>);
    const instance = test.instance();
    instance.setUsername('user1');
    expect(test.state('username')).toBe('user1');
});

test('Password handler sets the state of password', () => {
    Enzyme.configure({adapter: new Adapter()})
    const test = shallow(<LogInComponent/>);
    const instance = test.instance();
    instance.setPassword('password123');
    expect(test.state('password')).toBe('password123');
});