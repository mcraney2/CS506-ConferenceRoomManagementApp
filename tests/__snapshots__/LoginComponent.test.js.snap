// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders LogInComponent snapshot as expected 1`] = `
<View>
  <Text
    styles={
      Object {
        "color": "#474747",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      }
    }
  >
    Log-In Information
  </Text>
  <TextInput
    allowFontScaling={true}
    collapsable={false}
    editable={true}
    multiline={true}
    onChangeText={[Function]}
    onContentSizeChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    placeholder="Username"
    rejectResponderTermination={true}
    style={
      Array [
        30,
      ]
    }
    testID="text"
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    collapsable={false}
    editable={true}
    multiline={true}
    onChangeText={[Function]}
    onContentSizeChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    style={
      Array [
        30,
      ]
    }
    testID="text"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 25,
          "justifyContent": "center",
          "paddingHorizontal": 25,
          "paddingVertical": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
          }
        }
      >
        Log-In
      </Text>
    </View>
  </View>
</View>
`;
